internal/relay/relayer.go

```diff
@@ -190,19 +172,53 @@ func (r *Relayer) processMessageTX(ctx context.Context, m *MessageTX) error {
 	return nil
 }
 
+func (r *Relayer) buildTxQuery(ctx context.Context, m *MessageTX) (string, error) {
+	queryLastHeight, err := r.getLastQueryHeight(ctx, m.QueryId)
+	if err != nil {
+		return "", fmt.Errorf("could not get last query height: %w", err)
+	}
+
+	var params neutrontypes.TransactionsFilter
+	if err = json.Unmarshal([]byte(m.TransactionsFilter), &params); err != nil {
+		return "", fmt.Errorf("could not unmarshal transactions filter: %w", err)
+	}
+	// add filter by tx.height (tx.height>n)
+	params = append(params, neutrontypes.TransactionsFilterItem{Field: TxHeight, Op: "gt", Value: queryLastHeight})
+
+	queryString, err := queryFromTxFilter(params)
+	if err != nil {
+		return "", fmt.Errorf("failed to process tx query params: %w", err)
+	}
+
+	return queryString, nil
+}
+
 // getLastQueryHeight returns last query height & no err if query exists in storage, also initializes query with height = 0  if not exists yet
-func (r *Relayer) getLastQueryHeight(queryID uint64) (uint64, error) {
-	height, err := r.storage.GetLastQueryHeight(queryID)
-	if err == leveldb.ErrNotFound {
-		err = r.storage.SetLastQueryHeight(queryID, 0)
+func (r *Relayer) getLastQueryHeight(ctx context.Context, queryID uint64) (uint64, error) {
+	height, found, err := r.storage.GetLastQueryHeight(queryID)
+	if err != nil {
+		return 0, fmt.Errorf("could not get last query height from storage: %w", err)
+	}
+	if !found {
+		height = uint64(0)
+
+		if r.cfg.InitialTxSearchOffset != 0 {
```

Comment:
this will also affect all new queries. imagine you have a running relayer watching and serving a query of yours, you've run it with the `RELAYER_INITIAL_TX_SEARCH_OFFSET` set to some value. then you've got a need register another query. you register it, neutron sends an event for this query, and, since it's a new query, the `r.storage.GetLastQueryHeight` will return you `found=false` and therefore this query will also be handled starting with `height = current_height - RELAYER_INITIAL_TX_SEARCH_OFFSET`. is this the desired behaviour?

Reply:
Yeah, I intended it like this. Not sure if it's 100% useful tho. It can be useful for limiting search to some sane values (at least approximately around trusting window), since there is no other way

Reply:
yeah, makes sense